Deprecate 14 on the type scale (#4649)
alphagov/govuk-frontend: CHANGELOG.md

This change was introduced in [pull request #4646: Deprecate `element` parameter on button component](https://github.com/alphagov/govuk-frontend/pull/4646)

> #### Stop using `govuk-body-xs`, `govuk-!-font-size-14` and '14' on the typography scale

Reviewer: we're trialling using 'type scale' instead of 'typography scale', so I'm changing for consistency across docs.

```suggestion
#### Stop using `govuk-body-xs`, `govuk-!-font-size-14` and '14' on the type scale
```

> We have deprecated '14' on the responsive typography scale as well as classes within GOV.UK Frontend which use 14:

Author: I'm gonna have a think about this. I want to retain a flat `14` somewhere in there to communicate that it's the 14 point on the scale rather than _necessarily_ 14px (cause it drops down to 12px on small screens). I don't know how actually valuable that is though. Aside: this would be simpler if we had t-shirt sizing aliases set up.

> We have deprecated '14' on the responsive typography scale as well as classes within GOV.UK Frontend which use 14:

Author: I've added something to try and clarify that 14 the scale value ~= 14px the font size. Let me know what you think.

> We have deprecated '14' on the responsive typography scale as well as classes within GOV.UK Frontend which use 14:

Reviewer 2: Ah I know it doesn't, but we kinda say it does with that [guidance example above](https://github.com/alphagov/govuk-frontend/pull/4649#discussion_r1469308411)

Then in our Sass documentation we call them points?

>Point from the spacing scale (the size as it would appear on tablet and above)

Maybe we embrace that and spell it out as the guidance does?

>We have deprecated 14 point (14px large screens, 12px small screens) on the GOV.UK Frontend responsive typography scale, including font override classes that use 14 point:

> We have deprecated '14' on the responsive typography scale as well as classes within GOV.UK Frontend which use 14:

Author: I've made a light amend to refer to the 14 point as "point 14" just because it feels more natural to me. I'm open to disagreements.

> We have deprecated 14 on the responsive typography scale as well as classes within GOV.UK Frontend which set text to 14px:

Reviewer: ```suggestion
We have deprecated 14 on the responsive type scale, as well as classes within GOV.UK Frontend which set text to 14px:
```

- `govuk-body-xs`
- `govuk-!-font-size-14`

> We will be removing these classes and 14 on the typography scale in a future major release (6.0.0). From the next major release, you will no longer be able to call the Sass mixins `govuk-font` or `govuk-font-size` with `$size` set to '14'.

Reviewer: ```suggestion
We will be removing these classes and 14 on the type scale in GOV.UK Frontend release v6.0.0. From the next major release, you will no longer be able to call the Sass mixins `govuk-font` or `govuk-font-size` with `$size` set to '14'.
```

> This change was introduced in [#4649: Deprecate 14 on the typography scale](https://github.com/alphagov/govuk-frontend/pull/4649)

Reviewer: I'm not suggesting we change the 'typography scale' here because that would potentially mean changing the PR title as well.

Author: I changed the PR title anyway 😎

Reviewer: Ah ok!
